File: templates/nodejs/src/api/routes/$$channel$$.js
const Router = require('hermesjs/lib/router');
const router = new Router();
const {{ channelName | camelCase }}Handler = require('../handlers/{{ channelName | kebabCase }}');
module.exports = router;
{% if channel.hasPublish() %}
  {%- if channel.publish().summary() %}
/**
 * {{ channel.publish().summary() }}
 */
  {%- endif %}
router.use('{{ channelName | toHermesTopic }}', async (message, next) => {
  try {
    await {{ channelName | camelCase }}Handler.{{ channel.publish().id() }}({message});
    next();
  } catch (e) {
    next(e);
  }
});

{%- endif %}
{%- if channel.hasSubscribe() %}
  {%- if channel.subscribe().summary() %}
/**
 * {{ channel.subscribe().summary() }}
 */
  {%- endif %}
router.useOutbound('{{ channelName | toHermesTopic }}', async (message, next) => {
  try {
    await {{ channelName | camelCase }}Handler.{{ channel.subscribe().id() }}({message});
    next();
  } catch (e) {
    next(e);
  }
});

{%- endif %}

File: templates/go-ws/handlers/$$channel$$.go
package handlers

import (
	"log"
)
{% if channel.hasPublish() %}
func {{ channel.publish().id() | upperFirst }}(path string, message []byte) {
  log.Println(path, "->", string(message))
}
{% endif %}
{%- if channel.hasSubscribe() %}
func {{ channel.subscribe().id() | upperFirst }}(path string, message []byte) {
  log.Println(path, "->", string(message))
}
{% endif %}
